ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/formatter/IdParsers.java

```diff
@@ -44,7 +50,18 @@ public Optional<Id> parse(final String idString) {
             if (!matcher.find()) {
                 return Optional.empty();
             }
-            return IdFormatters.original().parse(idString);
+
+            val parserType = matcher.group(3);
+            if (parserType == null) {
+                return IdFormatters.original().parse(idString);
+            }
+
+            val parser = parserRegistry.get(Integer.parseInt(matcher.group(3)));
```

Comment:
What if matched group(3) is not integer parseable ?

Reply:
If the Id string doesn't have integer in group 3, the pattern will not match and it will return from line 51, As regex pattern has integer in group 3.